### PR TITLE
test: Spectal assumption for resolved reference equivalence

### PR DESCRIPTION
**Pull request by hudlow**
## PR summary

Add a test for the assumption that when Spectral resolves a schema, two identical `$ref` references will result in the same (`===` equivalent) in-memory object.

### General checklist
Please make sure that your PR fulfills the following requirements:  
- [x] The commit message follows the [Angular Commit Message Guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines).
- [x] Tests for the changes have been added (for bug fixes / features)

### CODE REVIEW

File: packages/ruleset/test/schema-resolution.test.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
+const { testRule } = require('./utils');
+
+describe('Spectral behavior assumptions', () => {
```
Comment (dpopp07) [inline packages/ruleset/test/schema-resolution.test.js:3]: If we want to contain any Spectral behavior assumption tests to this suite, which might be a good idea, we could rename this file and use it for any future tests too

**Review by dpopp07 — APPROVED** [review]
Looks good! 👍

**Review by padamstx — APPROVED** [review]
LGTM
I agree with @dpopp07 that this new testcase could be a container for all the tests involving Spectral behavior assumptions.  We could either rename it now or when (if?) we need to add more tests.

**Review by dpopp07 — APPROVED** [review]
Looks good! 👍

**Pull request merged**